browser/static/js/adjust.js

```diff
@@ -1,277 +1,376 @@
 // helper functions
 
-// modify image data in place to recolor
-function recolorScaled(data, i, j, jlen, r=255, g=255, b=255) {
-  // location in 1D array based on i and j
-  let pixel_num = jlen*j + i;
-  // set to color by changing RGB values
-  // data is clamped 8bit type, so +255 sets to 255, -255 sets to 0
-  data[(pixel_num*4)] += r;
-  data[(pixel_num*4) + 1] += g;
-  data[(pixel_num*4) + 2] += b;
-}
+class ImageAdjuster{
+  constructor(width, height, rgb, channelMax) {
+    // canvas element used for image processing
+    this.canvas = document.createElement('canvas');
+    this.canvas.id = 'adjustCanvas';
 
-// image adjustment functions: take img as input and manipulate data attribute
-// pixel data is 1D array of 8bit RGBA values
-function contrast_image(img, contrast = 0, brightness = 0) {
-  let d = img.data;
-  contrast = (contrast / 100) + 1;
-  for (let i = 0; i < d.length; i += 4) {
-      d[i] = d[i]*contrast + brightness;
-      d[i + 1] = d[i+1]*contrast + brightness;
-      d[i + 2] = d[i+2]*contrast + brightness;
-  }
-  return img;
-}
+    // this canvas should never be seen
+    this.canvas.style.display = 'none';
 
-function grayscale(img) {
-  let data = img.data;
-  for (var i = 0; i < data.length; i += 4) {
-      var avg = (data[i] + data[i + 1] + data[i + 2]) / 3;
-      data[i]     = avg; // red
-      data[i + 1] = avg; // green
-      data[i + 2] = avg; // blue
-    }
-  return img;
-}
+    // same dimensions as true image size
+    this.canvas.height = height;
+    this.canvas.width = width;
+    document.body.appendChild(this.canvas);
 
-function invert(img) {
-  let data = img.data;
-  for (var i = 0; i < data.length; i += 4) {
-    data[i]     = 255 - data[i];     // red
-    data[i + 1] = 255 - data[i + 1]; // green
-    data[i + 2] = 255 - data[i + 2]; // blue
-    }
-  return img;
-}
+    this.ctx = this.canvas.getContext("2d");
+    this.ctx.imageSmoothingEnabled = false;
+
+    // these will never change once initialized
+    this.height = height;
+    this.width = width;
+
+    this.rgb = rgb;
+
+    // TODO: also want invertMap for better adjustments
+    // when toggling between different channels
+    this.displayInvert = true;
 
-function preCompositeLabelMod(img, h1, h2) {
-  let ann = img.data;
-  // use label array to figure out which pixels to recolor
-  for (var j = 0; j < seg_array.length; j += 1){ //y
-    for (var i = 0; i < seg_array[j].length; i += 1){ //x
-      let jlen = seg_array[j].length;
-      let currentVal = Math.abs(seg_array[j][i]);
-      if (currentVal === h1 || currentVal === h2) {
-        recolorScaled(ann, i, j, jlen, r=255, g=-255, b=-255);
+    // brightness and contrast adjustment
+    // TODO: have this returned by a getter w/o a setter?
+    this.MIN_CONTRAST = -100;
+    this.MAX_CONTRAST = 700;
```

Comment:
I think the getter is a great idea. If a getter is used, this variable should be refactored to `_maxContrast` or something.